vpr/src/route/router_lookahead_map_utils.cpp

```diff
@@ -1432,22 +1431,20 @@ static void adjust_rr_pin_position(const RRNodeId rr, int& x, int& y) {
     VTR_ASSERT_SAFE(rr_graph.node_xlow(rr) == rr_graph.node_xhigh(rr));
     VTR_ASSERT_SAFE(rr_graph.node_ylow(rr) == rr_graph.node_yhigh(rr));
 
-    x = rr_graph.node_xlow(rr);
-    y = rr_graph.node_ylow(rr);
+    int x = rr_graph.node_xlow(rr);
+    int y = rr_graph.node_ylow(rr);
 
     /* Use the first side we can find
      * Note that this may NOT return an accurate coordinate
      * when a rr node appears on different sides
      * However, current test show that the simple strategy provides
      * a good trade-off between runtime and quality of results
      */
-    e_side rr_side = NUM_SIDES;
-    for (const e_side& candidate_side : SIDES) {
-        if (rr_graph.is_node_on_specific_side(rr, candidate_side)) {
-            rr_side = candidate_side;
-            break;
-        }
-    }
+    auto it = std::find_if(SIDES.begin(), SIDES.end(), [&](const e_side candidate_side) {
+        return rr_graph.is_node_on_specific_side(rr, candidate_side);
+    });
+
+    e_side rr_side = (it != SIDES.end()) ? *it : NUM_SIDES;
```

Comment:
Please add a comment above the .h declarations of these functions. This one is adjusting the pin location slightly to match the wires it can reach based on which side of the block it is at.